src/terminator/mod.rs

```diff
@@ -367,31 +367,43 @@ impl<'a, 'tcx> EvalContext<'a, 'tcx> {
 
         // Only trait methods can have a Self parameter.
 
+        // Intercept some methods (even if we can find MIR for them)
+        if let ty::InstanceDef::Item(def_id) = instance.def {
+            match &self.tcx.item_path_str(def_id)[..] {
+                "std::sys::imp::fast_thread_local::register_dtor" => {
```

Comment:
Because it evaluates the static the first time it is accessed. There is no miri support for weak linkage, in fact I'd be surprised if many people knew it turns the static into a pointer behind the scenes.